docker-compose.yml

```diff
@@ -0,0 +1,79 @@
+version: "3.8"
+
+services:
+  frontend:
+    container_name: frontend
+    build:
+      context: ./frontend
+    ports:
+      - 3000:3000
+    depends_on:
+      [user-service, matching-service, communication-service, collaboration-service, question-service]
+
+  user-service:
+    container_name: user-service
+    build:
+      context: ./user-service
+    ports:
+      - 8000:8000
+    depends_on:
+      [mongo-user, redis]
+
+  matching-service:
+    container_name: matching-service
+    build:
+      context: ./matching-service
+    ports:
+      - 8001:8001
+
+  communication-service:
+    container_name: communication-service
+    build:
+      context: ./communication-service
+    ports:
+      - 8002:8002
+
+  collaboration-service:
+    container_name: collaboration-service
+    build:
+      context: ./collaboration-service
+    ports:
+      - 8003:8003
+
+  question-service:
+    container_name: question-service
+    build:
+      context: ./question-service
+    ports:
+      - 8004:8004
+    depends_on:
+      - mongo-question
+
+  mongo-user:
+    container_name: mongo-user
+    image: mongo
+    ports:
+      - 27017:27017
+    volumes:
+      - mongo-user-volume:/data/db
```

Comment:
Why do we need volumes here? Are we not storing the data on mongoDB cloud database?

Reply:
```
const mongoDB = process.env.ENV == 'PROD' 
  ? process.env.DB_CLOUD_URI 
  : process.env.DB_LOCAL_URI || process.env.DB_DOCKER_URI;
```
With reference to the condition above, using database in Docker container is mainly to ease the development process as now, no installations/setups are required. As for deployment, I'm not too sure. Same applies to the related questions below.

Reply:
I think for now it's okay ba, but personally I feel it's better to redirect read and write to remote database so there's only a single source of truth.